types/src/task_state.rs

```diff
@@ -135,12 +136,29 @@ impl Task<Create> {
 
         // check input fkeys
         let inputs_spec: HashSet<&String> = function.inputs.iter().map(|f| &f.name).collect();
-        let req_input_fkeys: HashSet<&String> = req_input_owners.keys().collect();
+        let mut req_input_fkeys: HashSet<&String> = req_input_owners.keys().collect();
+        // If an input/output file is marked with `optional: True`, users do not need to
+        // register the file.
+        let option_inputs_spec: HashSet<&String> = function
+            .inputs
+            .iter()
+            .filter(|f| f.optional)
+            .map(|f| &f.name)
```

Comment:
checkout filter_map https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map

Reply:
I cannot come up with better ways of rewriting the code with `filter_map`. Do we have to replace it with the `filter_map`?
```Rust
let option_inputs_spec: HashSet<&String> = function
            .outputs
            .iter()
            .filter_map(|f| if f.optional { Some(&f.name) } else { None })
            .collect();
```

Reply:
nsdd. pass.